lib/generators/express_admin/scaffold/scaffold_generator.rb

```diff
@@ -0,0 +1,85 @@
+require 'rails/generators/erb/scaffold/scaffold_generator'
+
+module ExpressAdmin
+  module Generators
+    class ScaffoldGenerator < Erb::Generators::ScaffoldGenerator
+      source_root File.expand_path("../templates", __FILE__)
+      attr_reader :project_name, :project_path, :view_path, :resource_class
+
+      def create_root_folder
+        @project_name = if Rails.application
+                          Rails.application.class.parent_name
+                        else
+                          class_path.first.classify
+                        end
+        @project_path = @project_name.underscore
+        @resource_class = "#{@project_name}::#{singular_name.camelize}".classify
+        @view_path = File.join('app/views', @project_path, 'admin', plural_name)
+        empty_directory File.join("app/views", @project_path, "admin", controller_file_path)
+      end
+
+      def copy_view_files
+        available_views.each do |view|
+          filename = filename_with_extensions(view)
+          template "#{view}.html.et.erb", File.join("app/views", @project_path, "admin", controller_file_path, filename)
+        end
+      end
+
+      def create_controller_file
+        template "controller/controller.rb", File.join("app/controllers", @project_path, "admin", "#{controller_file_path}_controller.rb")
+      end
+
+      def create_model_file
+        template "model/model.rb", File.join("app/models/", @project_path, "#{singular_name}.rb")
+      end
+
+      def create_datatables_files
+        template "datatables/datatables.rb", File.join('app/datatables', @project_path, "#{singular_name}_datatable.rb")
+        template "datatables/datatables.js", File.join('app/assets/javascripts', @project_path, 'admin' ,"#{controller_file_name}.js")
+      end
+
+      def add_route
+        if File.readlines('config/routes.rb').grep(/Kernel\.const_defined\?\(\'ExpressAdmin::Engine\'\)/).any?
+          inject_into_file 'config/routes.rb', "          resources :#{controller_file_name}\n", after: /module: scope '#{@project_path}', as: '#{@project_path}' do\n/
+        else
+          append_to_file 'config/routes.rb',"\n\n
+if Kernel.const_defined?('ExpressAdmin::Engine')
```

Comment:
To make this work on the parent app without explicitly including `express_admin` in the parent app's `Gemfile`, I had to do two things:

1.  I had to upload `express_admin` to our private gem server in order for it to be included in the engine's `gemspec`. The source to the private gem server was added to the engine's `Gemfile`.
2.  I had to add the line
    `require 'express_admin'`
    in the `lib/engine_name/engine.rb` file.